src/jsonpatcherproxy.js

```diff
@@ -136,7 +136,11 @@ const JSONPatcherProxy = (function() {
       }
       operation.op = 'add';
       if (tree.hasOwnProperty(key)) {
-        if (typeof tree[key] !== 'undefined' || isTreeAnArray) {
+        if (typeof tree[key] !== 'undefined') {
+          if (tree[key] == newValue)
+            return true; // Value wasn't actually changed, pretend set was successful but don't generate a patch
```

Comment:
Wrap the `return true` statement in curly braces for readability

Reply:
Good point. Done!